### PR TITLE
Merging unified dev

### PR DESCRIPTION
**Pull request by addev-ashish**
Started with a rebase, which gave up towards the end. Had to merge the last 18 commits. The merge chart looks much better. Please let me know if you want another attempt at rebase.

To verify i compared pure merge with this and all the changes are same. All UTs passed.

### CODE REVIEW

File: adal/build.gradle [hunk 1]
```diff
@@ -138,44 +150,6 @@ task jacocoTestReport(type: JacocoReport, dependsOn: 'testDebugUnitTest') {
     ])
 }
 
-task createPom  {
```
Comment (heidijinxujia) [inline adal/build.gradle]: We will need the Pom file for the official release in both public maven and vsts maven. Not sure why we remove the pom task here.

File: adal/src/androidTest/java/com/microsoft/aad/adal/AuthenticationContextTest.java [hunk 2]
```diff
@@ -1096,7 +1115,8 @@ public void testScenarioLoginHintIdTokenDifferent() throws IOException, Authenti
         clearCache(context);
     }
 
-    @Test
+
+    @Ignore
```
Comment (heidijinxujia) [inline adal/src/androidTest/java/com/microsoft/aad/adal/AuthenticationContextTest.java]: Any context that why we want to skip this unit test?

File: adal/src/androidTest/java/com/microsoft/aad/adal/AuthenticationExceptionTests.java [hunk 3]
```diff
@@ -25,6 +25,9 @@
 
 import android.support.test.runner.AndroidJUnit4;
 
+
```
Comment (heidijinxujia) [inline adal/src/androidTest/java/com/microsoft/aad/adal/AuthenticationExceptionTests.java]: Extra lines

File: adal/src/androidTest/java/com/microsoft/aad/adal/BrokerAccountServiceTest.java [hunk 4]
```diff
@@ -42,6 +42,10 @@
 import android.util.Base64;
 import android.util.Pair;
 
+
```
Comment (heidijinxujia) [inline adal/src/androidTest/java/com/microsoft/aad/adal/BrokerAccountServiceTest.java]: nit: extra lines.

File: adal/src/androidTest/java/com/microsoft/aad/adal/DRSMetadataRequestorTests.java [hunk 5]
```diff
@@ -25,6 +25,10 @@
 
 import android.support.test.runner.AndroidJUnit4;
 
+
```
Comment (heidijinxujia) [inline adal/src/androidTest/java/com/microsoft/aad/adal/DRSMetadataRequestorTests.java]: nit: extra lines.

File: adal/src/androidTest/java/com/microsoft/aad/adal/DiscoveryTests.java [hunk 6]
```diff
@@ -28,6 +28,10 @@
 import android.support.test.filters.SdkSuppress;
 import android.support.test.runner.AndroidJUnit4;
 
+
```
Comment (heidijinxujia) [inline adal/src/androidTest/java/com/microsoft/aad/adal/DiscoveryTests.java]: Nit: extra lines.

File: adal/src/androidTest/java/com/microsoft/aad/adal/DefaultTokenCacheStoreTests.java [hunk 7]
```diff
@@ -29,6 +29,9 @@
 import android.support.test.InstrumentationRegistry;
 import android.support.test.runner.AndroidJUnit4;
 
+
```
Comment (heidijinxujia) [inline adal/src/androidTest/java/com/microsoft/aad/adal/DefaultTokenCacheStoreTests.java]: Nit: extra lines. We could use Ctrl + L in Android Studio to format the code automatically.

**Comment by addev-ashish** [comment]
@heidijinxujia these are all commits coming over from unified/dev. We will never finish this review :) unless you see me introducing them while they are not present in either branch, can you please file bugs for these (specially pom and ignored unit test)?

**Comment by iambmelt** [comment]
My test pass checked out. Ran interactive login (w/ SSO state sync to common cache) + silent auth. Everything appears working with updated redirect 👍

File: adal/build.gradle (hunk 1 above)
Comment (heidijinxujia) [reply adal/build.gradle]: We will fix this in #1185

**Pull request merged**